examples/plotting/zoom_map_example.py

```diff
@@ -0,0 +1,62 @@
+"""
+===================================
+zoom-out a particular region of map
+===================================
+How to plot a zoom-out image using matplotlib.
+"""
+import matplotlib.pyplot as plt
+from matplotlib.patches import ConnectionPatch
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE
```

Comment:
I think we should use a different sample file. I still think we should make this example look like the figure in the original issue.

Reply:
Hey @nabobalis thanks for the reviews i will make the changes. Also, Could you please look at this [comment](https://github.com/sunpy/sunpy/pull/4920#discussion_r567300217) from @wtbarnes
Maybe this can give a clear picture of what needed to be done. I am going with Will's wish it seems : )
Wow isn't Sun so beautiful with it's flares.




![zoomed](https://user-images.githubusercontent.com/72349558/107850458-a28f9c80-6e28-11eb-9468-9d9ccd12b09b.png)

Reply:
I still think the original image from the issue should be replicated in this example using the HMI sample data.

Reply:
Okay @nabobalis please guide me through it.

Reply:
What would you need guiding with?

Reply:
I just need to replace the AIA sample image with HMI sample image right?

Reply:
To be honest that will look better thanks! Will do it right now!

Reply:
That is one part. I think that original image from the issue is what we should be aiming to create in this example. Unless there is a reason not to?

Reply:
Where to find that image. I go through the SunPy documentation(Maybe could have missed as well). but i didn't find anything like that image.

Reply:
I asked Will about it long back it seems. Should i ask him again for that particular image?

Reply:
I am referring to the original PNG in the issue: https://user-images.githubusercontent.com/8676141/104660059-d776cb00-5693-11eb-84fd-89f94b690b60.png

The HMI image Will would have used is not in sunpy but we have a different HMI sample data image we can use. 
It's import is: "sunpy.data.sample.HMI_LOS_IMAGE"

Reply:
Are we expecting this kinda output (i will correct the connection path). Just making sure if this is the image i was supposed to use?



![Screenshot from 2021-02-13 20-24-07](https://user-images.githubusercontent.com/72349558/107853065-f656b180-6e39-11eb-8f1a-94b7df277b2e.png)

Reply:
Its closer, but still lacks the same colour and axis style as the original PNG example.

Reply:
Ok Nabil, if this is the expected image, i will start working on it.

Is there any way i can see `HMI_LOS_IMAGE` images. I just imported it, maybe i am not thinking clearly (there's just one HMI map image? ). 

Or maybe i can get them from below linked page?
https://docs.sunpy.org/en/stable/code_ref/data.html#module-sunpy.data.sample

Reply:
There is only one image, you can open it in a map and plot it. But this looks like the image I would expect.

Reply:
Hey @nabobalis running this example i.e.  `examples/plotting/hmi_synoptic_maps.py`

end up giving value error:

ValueError: 'Sine Latitude' did not parse as unit: At col 0, Sine is not a valid unit.  If this is meant to be a custom unit, define it with 'u.def_unit'. To have it recognized inside a file reader or other code, enable it with 'u.add_enabled_units'. For details, see https://docs.astropy.org/en/latest/units/combining_and_defining.html

tho i tried enabling it using some commands from [this link](https://docs.astropy.org/en/latest/units/combining_and_defining.html)  and `u.add_enabled_units` .
are we supposed to define Sine? or some package installation is missing. I have install them all i guess whatever was mentioned in documentation. 

I am not getting why zoomed image is lighter than real image. so thought of going through other example.

Reply:
Also, i was thinking (just an opinion) if we could keep both the images i.e. AIA and HMI 



![zoomed1](https://user-images.githubusercontent.com/72349558/107882791-64ba7300-6f11-11eb-8688-d2caf40f0ea9.png)

Reply:
> Hey @nabobalis running this example i.e. `examples/plotting/hmi_synoptic_maps.py`
> 
> ended up giving value error:
> 
> ValueError: 'Sine Latitude' did not parse as unit: At col 0, Sine is not a valid unit. If this is meant to be a custom unit, define it with 'u.def_unit'. To have it recognized inside a file reader or other code, enable it with 'u.add_enabled_units'. For details, see [docs.astropy.org/en/latest/units/combining_and_defining.html](https://docs.astropy.org/en/latest/units/combining_and_defining.html)
> 
> tho i tried enabling it using some commands from [this link](https://docs.astropy.org/en/latest/units/combining_and_defining.html) and `u.add_enabled_units` .
> are we supposed to define Sine? or some package installation is missing. I have install them all i guess whatever was mentioned in documentation.

I am not sure what is going on, as far as I can tell that example works. It could be that somehow the file downloaded in the example wasn't complete but otherwise I would make sure that astropy is up to date.

> I am not getting why zoomed image is lighter than real image. so thought of going through other examples.

I assume the data min/max values are different in the second image.

Reply:
> Also, i was thinking (just an opinion) if we could keep both the images i.e. AIA and HMI
> 
> ![zoomed1](https://user-images.githubusercontent.com/72349558/107882791-64ba7300-6f11-11eb-8688-d2caf40f0ea9.png)

I am not sure we need two images in the example. But I am open to being convinced.

Reply:
There's is no such value provided as max/min. Also i remember last time you mentioned why code isn't that complex as given in real issue. So i thought i will let this as it is (have updated the comments). 
Will create a new example using that broken code and HMI map image. 

Also if there's requirement of two different code of zoomed in plotting with two different image then will commit AIA zoomed image example as well.
Is this okay to go with Nabil?

Reply:
No, I think the plot command itself will autoscale somewhat when you plot it.

If you want to create two examples and two PRs, the choice is yours.

Reply:
I found two kinda HMI image one lighter than the other!! i used these [codes](https://docs.sunpy.org/en/latest/_downloads/ae19f8bd004e215d47e15a54c20930df/downloading_hmi.py) to download it.




![Screenshot from 2021-02-14 22-48-14](https://user-images.githubusercontent.com/72349558/107883783-009aad80-6f17-11eb-9aee-35ba28e36f59.png)

Reply:
Sure but we should use the sample data one since we don't have to download it each time for the example.

Reply:
I agree but doesn't it seems like that zoomed image is child of this image 😀
will figure it out i guess.

Reply:
You should be able to set the vmin/vmax as keywords args to plot.

Reply:
Hey @nabobalis I was to ask you, should i use magnetogram code to plot HMI map image or simple coding is fine. I think magnetogram one should be used as this image itself is showing strength of magnetic field. 

As i am facing lots of problem related with code and diving deep into it. Maybe i will need some more time.
My two pending PR #4920 and PR #4867 are ready to review i believe!

Should we close this two PR and add one later on when done with coding?
Can i make two PR on one issue? or maybe i should do one more commit? asking you! as you say will do.

This much density is okay Nabil? (I have set vmin=-1e2, and  vmax=1e2) 

also please ignore the alignment of labels and connecting lines
 (code is not working properly i need to figure out these stuffs asap)


![HMI](https://user-images.githubusercontent.com/72349558/108519569-c6068b80-72ef-11eb-92c8-e5d4e0a611c1.png)

Reply:
I don't quite understand what you mean by "magnetogram code or simple coding".

I also don't understand why you want to close the two open pull requests? 

You can open as many pull requests as needed for an issue. There are no fixed rules.

The image looks ok.

Reply:
I am sorry my question was little vague! i mean should i use these [codes](https://github.com/sunpy/sunpy/issues/4874#issue-786408684)

or the one i used for zoomed-in plotting of AIA image! or anything is fine?

> I also don't understand why you want to close the two open pull requests?



oh okay i thought we are close to release of 2.1 so maybe need to finished them up asap. Sorry acting like a kid! yeah i got it.

Reply:
> I am sorry my question was little vague! i mean should i use these [codes](https://github.com/sunpy/sunpy/issues/4874#issue-786408684)
> 
> or the one i used for zoomed-in plotting of AIA image! or anything is fine?
> (i remember you mentioning why code isn't that complex as mentioned in issue? so i thought i will ask you)

Again I don't quite understand, I think it's worth just ignoring it and we will focus on this example as it is. 

> oh okay i thought we are close to release of 2.1 so maybe need to finished them up asap. Sorry acting like a kid! yeah i got it.(don't know much about these stuffs)

We are close but these PRs will not block the release so there is no rush.

Reply:
Okay got it! Also should i continue working on zoom plotting using HMI map image? I think i am quite close.

Reply:
Up to you, I don't think we need two seperate examples doing the same thing.

Reply:
I agree i will drop the idea then! It should be based on requirement rather than a person wish : )